src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -364,70 +298,83 @@
      * Checks whether the given stack is contained in this Inventory.
      * The stack size is ignored. Note this will return true if any amount
      * of the supplied stack is found. To check if an inventory contains at
-     * least an amount use {@link #contains(ItemStack)}.
+     * least a given quantity use {@link #contains(ItemStack)}.
      *
      * @param stack The stack to check for
+     *
      * @return True if the stack is present in this inventory
      */
     boolean containsAny(ItemStack stack);
 
+    // TODO remove from API? do we need to get a property relative to another parent in API?
```

Comment:
Agreed. if they want the parent? Call getParent().getProperty().

Should be only one of those.

Reply:
Thats not the same. Properties of a child are relative to its parent.
The way queries work there can be multiple parents returning the "same" (in terms of the same lens not instance) child.

Reply:
Right but again there is no reason to call a method to get one from the parent. You could just get the Parent inventory and call it instead no?

Reply:
See https://github.com/SpongePowered/SpongeCommon/issues/2378#issuecomment-536341811. Each slot in a chest has three `SlotIndex` properties: one for the index within the grid, one for the index within the row, and one for the index within the column. Each is accesed by specifying a different parent inventory: the grid, the row, or the column.

Reply:
Although personally, I think `SlotIndex` (and other properties that depend on a parent, such as `SlotPos`) ought to be methods instead of properties - that way you have to specify a parent. For example, `SlotIndex` could be replaced by `Inventory.getSlot(int index)` and `Inventory.indexOf(Inventory inv)`.